contracts/utils/Ownable.sol

```diff
@@ -4,39 +4,84 @@ pragma solidity ^0.8.0;
 
 import { IOwnable } from '../interfaces/IOwnable.sol';
 
+/**
+ * @title Ownable
+ * @notice A contract module which provides a basic access control mechanism, where
+ * there is an account (an owner) that can be granted exclusive access to
+ * specific functions.
+ *
+ * The owner account is set through ownership transfer. This module makes
+ * it possible to transfer the ownership of the contract to a new account in one
+ * step, as well as to an interim pending owner. In the second flow the ownership does not
+ * change until the pending owner accepts the ownership transfer.
+ */
 abstract contract Ownable is IOwnable {
     // keccak256('owner')
     bytes32 internal constant _OWNER_SLOT = 0x02016836a56b71f0d02689e69e326f4f4c1b9057164ef592671cf0d37c8040c0;
     // keccak256('ownership-transfer')
     bytes32 internal constant _OWNERSHIP_TRANSFER_SLOT =
         0x9855384122b55936fbfb8ca5120e63c6537a1ac40caf6ae33502b3c5da8c87d1;
 
+    /**
+     * @notice Modifier that throws an error if called by any account other than the owner.
+     */
     modifier onlyOwner() {
         if (owner() != msg.sender) revert NotOwner();
 
         _;
     }
 
+    /**
+     * @notice Returns the current owner of the contract.
+     * @return owner_ The current owner of the contract
+     */
     function owner() public view returns (address owner_) {
         assembly {
             owner_ := sload(_OWNER_SLOT)
         }
     }
 
+    /**
+     * @notice Returns the pending owner of the contract.
+     * @return owner_ The pending owner of the contract
+     */
     function pendingOwner() public view returns (address owner_) {
         assembly {
             owner_ := sload(_OWNERSHIP_TRANSFER_SLOT)
         }
     }
 
+    /**
+     * @notice Transfers ownership of the contract to a new account `newOwner`.
+     * @dev Can only be called by the current owner.
+     * @param newOwner The address to transfer ownership to
+     */
     function transferOwnership(address newOwner) external virtual onlyOwner {
+        emit OwnershipTransferred(newOwner);
+
+        assembly {
+            sstore(_OWNER_SLOT, newOwner)
```

Comment:
```suggestion
            sstore(_OWNER_SLOT, newOwner)
            sstore(_OWNERSHIP_TRANSFER_SLOT, 0)
```
pending owner should also be reset so it can't accept after a switch

Reply:
makes sense